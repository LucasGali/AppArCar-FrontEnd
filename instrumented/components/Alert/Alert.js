import React, { useState, useEffect } from 'react';

function Alert(props) {
  const [modalDisplay, toggleDisplay] = useState('none');
  const openModal = () => {
    toggleDisplay('block');
  };
  const closeModal = () => {
    toggleDisplay('none');
    props.hideError(null);
  };
  useEffect(() => {
    if (props.errorMessage !== null) {
      openModal();
    } else {
      closeModal();
    }
  });

  return (
    <div
      className={'alert alert-danger alert-dismissable mt-4 alert-msg'}
      role="alert"
      id="alertPopUp"
      style={{ display: modalDisplay }}
    >
      <div className="d-flex">
        <span data-testid="alert-msg">{props.errorMessage}</span>
        <button
          type="button"
          className="close ml-2"
          aria-label="Close"
          data-testid="alert-btn"
          onClick={() => closeModal()}
        >
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
    </div>
  );
}

export default Alert;
