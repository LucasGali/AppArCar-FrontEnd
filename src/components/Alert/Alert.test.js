import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Alert from './Alert';
import '@testing-library/jest-dom/extend-expect';

describe('Alert', () => {
  const errorMessage = 'This is an error';

  it('render content successfully', () => {
    render(<Alert errorMessage={errorMessage} />);
    const msg = screen.getByTestId('alert-msg')
    expect(msg).toBeInTheDocument();
    expect(msg).toHaveTextContent('This is an error');
  });

  it('check if there is a btn inside the modal', () => {
    const { getByTestId } = render(
      <Alert errorMessage={errorMessage} />,
    );
    expect(getByTestId('alert-btn')).toBeInTheDocument();
  });

  it('check if fn is trigger on click to close modal', () => {
    const spyFunction = jest.fn();
    const { getByTestId } = render(
      <Alert errorMessage={errorMessage} hideError={spyFunction} />,
    );
    fireEvent.click(getByTestId('alert-btn'));
    expect(spyFunction).toHaveBeenCalledTimes(1);
    expect(spyFunction).toHaveBeenCalledWith(null);
   });

  it('snapshot comparison', () => {
    const { asFragment } = render(<Alert errorMessage={errorMessage} />);
    expect(asFragment(<Alert />)).toMatchSnapshot();
  });
});
