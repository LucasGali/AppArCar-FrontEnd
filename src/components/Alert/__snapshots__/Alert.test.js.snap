// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshot comparison 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger alert-dismissable mt-4 alert-msg"
    id="alertPopUp"
    role="alert"
    style="display: block;"
  >
    <div
      class="d-flex"
    >
      <span
        data-testid="alert-msg"
      >
        This is an error
      </span>
      <button
        aria-label="Close"
        class="close ml-2"
        data-testid="alert-btn"
        type="button"
      >
        <span
          aria-hidden="true"
        >
          Ã—
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
